sklearn/decomposition/fastica_.py

```diff
@@ -495,14 +519,23 @@ def _fit(self, X, compute_sources=False):
             X_new : array-like, shape (n_samples, n_components)
         """
         fun_args = {} if self.fun_args is None else self.fun_args
+
+        whiten = self.whiten
+        if whiten is True:
```

Comment:
I guess this validation could be done in `fastica()` and not here. That function also needs to understand `unit-variance` if we're changing the default here. And any module using that function should get the same warning I suppose.

Reply:
I followed the guide from [here](https://scikit-learn.org/stable/developers/contributing.html#deprecation) (the fourth paragraph), and makes sense to me because `unit-variance` and `True` are they behave in the same way for `fastica()`. Of course, I only know about one or two modules of the project, so if you say so that I'll change it, but I wanted to point that out.

Also, `fastica()` is only used by the tests and by `FastICA` class. Would it then make sense then to change `fastica()` to a private function?